### PR TITLE
COMMERCE-5273 TooltipPrice renderer converted to digest data from Headless API

### PR DESCRIPTION
**Pull request by FabioDiegoMastrorilli**
The old TooltipPrice data renderer could not work with headless API because It needed them to provide an object build the way the component was expecting. 
With the new implementation, the table entries can be defined via the data set display views schemas.

Also, the component should be able to use different renderers for displaying data in a row (please see the discount row):
image

Due to avoid cycle dependencies, I moved some utility functions from `data_set_display/data_renderers/index.js` to `data_set_display/utilities/dataRenderers.js`

Documentation for most of the renderers will come soon.

### CODE REVIEW

**Comment by FabioDiegoMastrorilli** [comment]
ci:test:sf

**Comment by FabioDiegoMastrorilli** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
ci:test:commerce

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/TooltipResumeRenderer.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+			);
+
+			if (value) {
+				return [
+					...rowsData,
+					{
+						...rowDefinition,
+						value,
+					},
+				];
```
Comment (wincent) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/TooltipResumeRenderer.js:80]: Once thing about this pattern is that you are copying the array on every step, producing a lot of short-lived intermediate objects that apply pressure on the garbage collector for no good reason. Seeing as you "own" the `rowsData` you can safely mutate it and avoid the copying. eg.

```js
const tooltipTableRows = options.details.rowsDefinitions.reduce(
	(rowsData, rowDefinition) => {
		const value = getSchemaString(itemData, rowDefinition.valueFieldKey);

		if (value) {
			rowsData.push({
				...rowDefinition,
				value,
			});
		}

		return rowsData;
	},
	[]
);
```

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/TooltipResumeRenderer.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+		<>
+			{value}
+			{tooltipTableRows.length && (
+				<ClayTooltipProvider
+					contentRenderer={() => (
+						<table className="tooltip-table">
+							<tbody>
+								{tooltipTableRows.map((rowData, i) => (
+									<TooltipTableRow key={i} {...rowData} />
+								))}
```
Comment (wincent) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/TooltipResumeRenderer.js:98]: [Integer keys aren't the best choice for React `key` props](https://reactjs.org/docs/lists-and-keys.html) if you have an alternative value that is unique or identifying. In this case, maybe `rowData.valueFieldKey` would be a better key?

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/utilities/dataRenderers.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+	return dataRenderer;
+}
+
+export const fetchedContentRenderers = [];
+
+export function getDataRendererByURL(url) {
+	return new Promise((resolve, reject) => {
+		const addedDataRenderer = fetchedContentRenderers.find(
+			(cr) => cr.url === url
```
Comment (wincent) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/utilities/dataRenderers.js:33]: [Abbreviations are bad](https://github.com/liferay/liferay-frontend-guidelines/blob/master/general/naming.md#avoid-abbreviations). Especially so for one like this which already has a widely-used meaning in computing (["carriage return"](https://en.wikipedia.org/wiki/Carriage_return)).

**Review by wincent — COMMENTED** [review]
lg2m. Some nit-comments inline that you can take or leave.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/TooltipResumeRenderer.js (hunk 2 above)
Comment (FabioDiegoMastrorilli) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/TooltipResumeRenderer.js:98]: Hi Greg, valueFieldKey might also be an array to define nested values such "workflowStatusInfo.label". I can use an integer or the `.join` method. Would you go for the second option?

Comment (wincent) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/TooltipResumeRenderer.js:98]: Nah. I wouldn't both with `join()`. I also wondered about `rowData.label`; is that going to be unique? If not, the number is fine.

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/ImageRenderer.js [hunk 4]
```diff
@@ -19,53 +19,31 @@ import React from 'react';
 import {getValueFromItem} from '../utilities/index';
 
 function ImageRenderer(props) {
+	const imgProps =
```
Comment (jbalsas) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/ImageRenderer.js:22]: Here's an abbreviation that escaped @wincent's keen 👀

Comment (jbalsas) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/ImageRenderer.js:22]: Maybe this one is so common that we don't mind it though

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/TooltipResumeRenderer.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+					itemData={itemData}
+					options={options}
+					value={value}
+				/>
+			</td>
+		</tr>
+	);
+}
+
+function TooltipResumeRenderer({itemData, options, value}) {
```
Comment (jbalsas) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/TooltipResumeRenderer.js:61]: Is `Tooltip**Resume**Renderer` a proper name for this? Would `TooltipSummaryRenderer` be more appropriate? When I hear "resume" I immediately think about "restart".

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/index.js [hunk 6]
```diff
@@ -25,7 +24,7 @@ import LinkRenderer from './LinkRenderer';
 import ListRenderer from './ListRenderer';
 import QuantitySelectorRenderer from './QuantitySelectorRenderer';
 import StatusRenderer from './StatusRenderer';
-import TooltipPriceRenderer from './TooltipPriceRenderer';
+import TooltipResumeRenderer from './TooltipResumeRenderer';
```
Comment (jbalsas) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/index.js:27]: At least this now sounds less commerce-specific, so 👍

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/ImageRenderer.js (hunk 4 above)
Comment (wincent) [reply modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/data_renderers/ImageRenderer.js:22]: [No exceptions](https://en.wikipedia.org/wiki/Slippery_slope)!

Except, of course, for [the exceptions](https://github.com/liferay/liferay-frontend-guidelines/blob/master/general/naming.md#examples-of-good-abbreviations).

**Comment by FabioDiegoMastrorilli** [comment]
@wincent and @jbalsas changes had been done

**Comment by FabioDiegoMastrorilli** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
CI isn't making it any easy today 😢

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
ci:forward

**Pull request closed without merging**